internal/rpcserver/rpcwebsocket.go

```diff
@@ -1466,18 +1463,10 @@ out:
 				break out
 			case !c.authenticated:
 				// Check credentials.
-				login := authCmd.Username + ":" + authCmd.Passphrase
-				auth := "Basic " + base64.StdEncoding.EncodeToString([]byte(login))
-				mac := make([]byte, 0, sha256.Size)
-				mac = c.rpcServer.authMAC(mac, []byte(auth))
-				cmp := subtle.ConstantTimeCompare(mac, c.rpcServer.authsha[:])
-				limitcmp := subtle.ConstantTimeCompare(mac, c.rpcServer.limitauthsha[:])
-				if cmp|limitcmp != 0 {
-					log.Warnf("Auth failure.")
+				c.authenticated, c.isAdmin = c.rpcServer.checkAuthUserPass(authCmd.Username, authCmd.Passphrase, c.addr)
```

Comment:
This is exceeding the 80-char limit imposed by the code contribution guidelines.